library/components/TTooltip.vue

```diff
@@ -1,64 +1,119 @@
 <template>
-  <div
-    class="p-4 bg-[#1C1C21] text-white w-max h-max rounded absolute top-0 z-50"
-    :style="{ width, ...tooltipPositions }"
-  >
-    <slot></slot>
-    <div
-		class="bg-[#1C1C21] w-4 h-4 absolute rotate-45 m-auto"
-		:class="arrowPositions[position]"
-	></div>
-  </div>
+	<div @mouseover="handleMouseOver" @click="handleClick" ref="tooltipContainer">
+		<div ref="trigger">
+			<slot name="trigger">
+				<help-circle-outline-icon :size="12" />
+			</slot>
+		</div>
+		<div
+			class="p-4 bg-[#1C1C21] text-white w-max h-max rounded fixed z-50"
+			:style="{ width, ...positions.tooltip }"
+			ref="tooltip"
+			v-show="showTooltip"
+			@mouseleave="handleMouseLeave"
+		>
+			<slot></slot>
```

Comment:
I think the default slot should be for the trigger element. And the message displayed in the tooltip should be an enforced prop. This would avoid being able to use the tooltip without a message: 

ProductCL has a good example: https://github.com/PatchUI/ProductCL/blob/main/src/components/BaseTooltip/BaseTooltip.vue

Reply:
The problem with using prop is that, it is only limited to text, but according to one of the slide I saw inside figma, our tooltips can also have clickable links, this was a big reason why I went with custom tooltips.

Reply:
Ah, ok. 

TIL: clickable content can be ok in a tooltip: https://www.nngroup.com/articles/tooltip-guidelines/ (section 3)